src/Faction/Factions.ts

```diff
@@ -38,5 +38,6 @@ export function loadFactions(saveString: string): void {
     if (loadedFaction.alreadyInvited) faction.alreadyInvited = true;
```

Comment:
I just saw this TODO note. @Snarling what's the plan here?

Reply:
Factions used to be a huge part of the savegame. For instance the FactionSave was lines 4 through 787 here: https://github.com/bitburner-official/bitburner-src/blob/1de676972f25fa265a0159bc233762e9310cfd8c/test/jest/__snapshots__/FullSave.test.ts.snap

Most of the information being saved was not actually needed in the save file, since it was static info. After some changes, much of the information was able to be removed from the savegame, which makes saving faster and savefiles smaller. Here the size of the factions save is shrunk by over half. https://github.com/bitburner-official/bitburner-src/blob/9a0a843ffc507a9d11d67d97d0010ff608df5c41/test/jest/__snapshots__/FullSave.test.ts.snap

But most of the remaining info is still duplicate info from the Player object, so I thought about ways we could deduplicate this stored info.

faction.alreadyInvited: player has a factionInvites array that has the same info
faction.isMember: player has a factions array with the same info
faction.isBanned: not currently stored on player. So this one probably needs to stay on the factions object.

Why do these need to be on the faction object right now?

Operations like checking if the player is in a particular faction are slow with an array of faction names (have to iterate and search for a matching member in array), but fast with a property on the faction. But with some data structure changes, the data on Player could be suitable for this kind of lookup as well.

e.g. changing player.factions to a set instead of array: `player.factions.has(factionName)` replaces `Factions[factionName].isMember`
player.factionInvites as a set instead of array: `player.factionInvites.has(factionName)` replaces `Factions[factionName].alreadyInvited`